File: tools/generate-screenshots.py
#!/usr/bin/env python3
"""
Generate screenshots of MkDocs site in light and dark mode using Playwright.
"""

import asyncio
from pathlib import Path
from playwright.async_api import async_playwright

OUT_DIR = Path("docs/assets/img")
OUT_DIR.mkdir(parents=True, exist_ok=True)

async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        context = await browser.new_context()
        page = await context.new_page()

        # Light mode
        await page.emulate_media(color_scheme="light")
        await page.goto("http://127.0.0.1:8000/")
        await page.screenshot(path=OUT_DIR / "preview-light.png", full_page=True)

        # Dark mode
        await page.emulate_media(color_scheme="dark")
        await page.goto("http://127.0.0.1:8000/")
        await page.screenshot(path=OUT_DIR / "preview-dark.png", full_page=True)

        await browser.close()

if __name__ == "__main__":
    asyncio.run(main())

File: tools/generate-screenshots.js
const fs = require("fs");
const path = require("path");
const puppeteer = require("puppeteer");

(async () => {
  const baseUrl = process.env.BASE_URL || "http://localhost:8080/";
  const outDir = path.resolve(".screenshots");
  if (!fs.existsSync(outDir)) fs.mkdirSync(outDir, { recursive: true });

  const modes = [
    { name: "dark", value: "dark" },
    { name: "light", value: "light" }
  ];

  const browser = await puppeteer.launch({ headless: "new" });
  const page = await browser.newPage();
  await page.setViewport({ width: 1440, height: 900 });

  for (const m of modes) {
    await page.emulateMediaFeatures([{ name: "prefers-color-scheme", value: m.value }]);
    await page.goto(baseUrl, { waitUntil: "networkidle0" });
    const file = path.join(outDir, `screenshot-${m.name}.png`);
    await page.screenshot({ path: file, fullPage: true });
    console.log(`Saved ${file}`);
  }

  await browser.close();
})().catch(err => {
  console.error(err);
  process.exit(1);
});
